src/scss/components/banner.scss

```diff
@@ -807,214 +826,110 @@
-          }
-        }
+    &:after {
+      background: rgba(0, 0, 0, .5);
+      @include container-query() {
+        background: linear-gradient(180deg, #{$gradient-dark-vertical});
       }
     }
 
-    &[class*="banner--gradient-dark"] {
-      .headline {
-        color: $secondary;
-        text-shadow: none;
-        @include breakpoint(sm) {
-          color: #fff;
-          text-shadow: 0 1px 1px #000;
-        }
-
-        a {
-          color: $secondary;
+    &.banner--gradient-light {
+      &[class*="banner--stacked"] {
+        &:after {
+          background: none;
+          width: auto;
+          height: auto;
           @include breakpoint(sm) {
-            color: #fff;
+            width: 100%;
+            height: 100%;
+            background: linear-gradient(180deg, #{$gradient-light-vertical});
           }
         }
+      }
 
-        [class*="bg--black"] & {
-          color: #fff;
-          text-shadow: 0 1px 1px #000;
-          @include breakpoint(sm) {
-            color: #fff;
-          }
-
-          a {
-            color: #fff;
-            @include breakpoint(sm) {
-              color: #fff;
-            }
-          }
+      &:after {
+        background: rgba(255, 255, 255, .7);
+        @include container-query() {
+          background: linear-gradient(180deg, #{$gradient-light-vertical});
         }
       }
     }
   }
 
-  &[class*="bg--gray"] {
-    .headline {
-      color: $secondary;
-      text-shadow: none;
-
-      a {
-        color: $secondary;
-      }
+  &--transparent {
+    &:after {
+      background: none;
     }
   }
 
-  &[class*="bg--black"] {
-    .headline {
-      a {
-        color: #fff;
-      }
-    }
+  &[class*="bg-"]:after {
+    background: inherit;
   }
+}
+// === End: Gradients === //
 
-  &[class*="bg--gold"],
-  &[class*="bg--white"],
-  &[class*="banner--gradient-light"]:not([class*="bg-"]) {
-    .headline {
-      color: $secondary;
-      text-shadow: none;
+// === Background colors === //
+// Remove left/right padding if container background matches banner background.
 
-      a {
-        color: $secondary;
+$bg-colors: "white", "black", "gray", "gold";
+
+// Simple background colors and pattern variants
+@each $color in $bg-colors {
+  div.bg--#{$color} {
+    .banner.bg--#{$color} {
+      .banner__content {
+        padding-left: 0;
+        padding-right: 0;
       }
     }
+  }
 
-    &.banner--stacked:not([class*="bg-"]) {
-      [class*="bg--black"] & {
-        .headline {
-          color: #fff;
-          @include breakpoint(sm) {
-            color: $secondary;
-          }
-
-          a {
-            color: #fff;
-            @include breakpoint(sm) {
-              color: $secondary;
-            }
-          }
-        }
+  // Pattern variant for each color.
+  div[class*="bg--#{$color}--pattern"] {
+    .banner[class*="bg--#{$color}--pattern"] {
+      .banner__content {
+        padding-left: 0;
+        padding-right: 0;
```

Comment:
I'm wondering if these two sections could be consolidated down to using one rule that uses a start of string match for `bg--#{$color}` instead of an exact match on the whole class name? This would allow the rule to match both patterns and solid colors, which was intended with the way those classes were named.